File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/personal/PersonalSignTest.java
package opensdk.sdk.apis.personal;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.PersonalSignResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

@DisplayName("Personal RPC Test")
public class PersonalSignTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC personal_sign")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        String message = "0xdead";
        String address = "0xb44b66f0d6ea803175f921018cba7e914fed25b9";
        String passphrase = "helloWorld";

        PersonalSignResponse response = sdk.personal.sign(message, address, passphrase)
                .send();
        assertNotNull(response);
        assertNull(response.getError());
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/personal/PersonalSign.test.js
const OpenSdk = require("opensdk-javascript");
const { expect } = require("@jest/globals");
const { RPC } = require("../constant");
const { unlockAccount } = require("../../helpers/eth");

const sdk = new OpenSdk(new OpenSdk.ApiClient(RPC));

describe('personal_sign API', () => {
    test('should return personal_sign', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined()
            expect(data).toMatch(/^0x[0-9a-fA-F]+$/)
            done();
        };
        unlockAccount().then(address => {
            const message = '0xdeadbeaf';
            const password = "helloWorld";

            sdk.personal.sign(message, address, password, {}, callbackOne);
        })
    });
});